Fix amlogic

CodeChenL opened the pull request:
关于amlogic系列的一揽子修复/优化

moonlinux committed:
arm64: dts: amlogic: Add cache information to the Amlogic G12A SoCS

As per the S905X2 datasheet add missing cache information to the Amlogic
G12A SoC.

- Each Cortex-A53 core has 32KB of L1 instruction cache available and
	32KB of L1 data cache available.
- Along with 512KB Unified L2 cache.

Cache memory significantly reduces the time it takes for the CPU
to access data and instructions, leading to faster program execution
and overall system responsiveness.

Signed-off-by: Anand Moon <linux.amoon@gmail.com>
Link: https://lore.kernel.org/r/20250825065240.22577-4-linux.amoon@gmail.com
Signed-off-by: Neil Armstrong <neil.armstrong@linaro.org>

moonlinux committed:
arm64: dts: amlogic: Add cache information to the Amlogic S922X SoC

As per S922X datasheet add missing cache information to the Amlogic
S922X SoC.

- Each Cortex-A53 core has 32 KB of instruction cache and
	32 KB of L1 data cache available.
- Each Cortex-A73 core has 64 KB of L1 instruction cache and
	64 KB of L1 data cache available.
- The little (A53) cluster has 256 KB of unified L2 cache available.
- The big (A73) cluster has 1 MB of unified L2 cache available.

Cache memory significantly reduces the time it takes for the CPU
to access data and instructions, leading to faster program execution
and overall system responsiveness.

Signed-off-by: Anand Moon <linux.amoon@gmail.com>
Link: https://lore.kernel.org/r/20250825065240.22577-11-linux.amoon@gmail.com
Signed-off-by: Neil Armstrong <neil.armstrong@linaro.org>

glaroque committed:
arm64: dts: amlogic: meson-g12b: Fix L2 cache reference for S922X CPUs

The original addition of cache information for the Amlogic S922X SoC
used the wrong next-level cache node for CPU cores 100 and 101,
incorrectly referencing `l2_cache_l`. These cores actually belong to
the big cluster and should reference `l2_cache_b`. Update the device
tree accordingly.

Fixes: e7f85e6c155a ("arm64: dts: amlogic: Add cache information to the Amlogic S922X SoC")
Signed-off-by: Guillaume La Roque <glaroque@baylibre.com>
Reviewed-by: Neil Armstrong <neil.armstrong@linaro.org>
Link: https://patch.msgid.link/20251123-fixkhadas-v1-1-045348f0a4c2@baylibre.com
Signed-off-by: Neil Armstrong <neil.armstrong@linaro.org>

CodeChenL committed:
arm64: dts: amlogic: meson-g12b: fix L1 cache sizes for S922X CPUs

Amlogic S922X SoC each Cortex-A73 core has 64 KB of L1 instruction cache and
64 KB of L1 data cache available.
The original addition of cache information for the Amlogic S922X SoC
used the wrong sizes of L1 cache.

Fixes: e7f85e6c155a ("arm64: dts: amlogic: Add cache information to the Amlogic S922X
SoC")

Signed-off-by: Jiali Chen <chenjiali@radxa.com>

CodeChenL committed:
arm64: dts: amlogic: g12: increase linux,cma size

Video codec requires more CMA memory

Signed-off-by: Jiali Chen <chenjiali@radxa.com>

CodeChenL committed:
arm64: dts: meson: radxa-zero2: add fcs,fusb302 and gpio-sbu-mux

To support Type-C automatic flipping

Signed-off-by: Jiali Chen <chenjiali@radxa.com>

CodeChenL committed:
arm64: dts: meson-g12a: modify cpu opp voltage value

Reference to vendor kernel [1].

[1] https://github.com/CoreELEC/linux-amlogic/blob/amlogic-4.9-20/arch/arm64/boot/dts/amlogic/g12a_s905x2_u211.dts

Signed-off-by: Jiali Chen <chenjiali@radxa.com>

CodeChenL committed:
arm64: qcom_module.config: add qcom_module.config

Because the qcom_module_defconfig with the deconfig suffix clears the arm64/deconfig configuration [1],
causing other manufacturers' SOC to fail to start,
a symbolic link is created to qcom_module.config for use by Radxa OS,
and can also share the configuration with other BSP operating systems.

[1] https://github.com/radxa/kernel/blob/linux-6.17.1/scripts/kconfig/Makefile#L102

Signed-off-by: Jiali Chen <chenjiali@radxa.com>

RadxaYuntian commented on arch/arm64/boot/dts/amlogic/Makefile:
```diff
@@ -36,6 +40,7 @@ dtb-$(CONFIG_ARCH_MESON) += meson-g12b-odroid-n2-plus.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-g12b-odroid-n2.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-g12b-odroid-n2l.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-g12b-radxa-zero2.dtb
+dtb-$(CONFIG_ARCH_MESON) += meson-g12b-radxa-zero-2pro.dtb
```
这个提交里面没有看到有软链接呀

CodeChenL replied:
多提交了
```suggestion
```

RadxaYuntian replied:
其实加一个单独提交，然后保留这个软链接也可以，跟现有的uboot兼容是吧？

CodeChenL replied:
软链接之后是兼容老uboot的

RadxaYuntian commented on arch/arm64/boot/dts/amlogic/meson-g12b-radxa-zero2.dts:
```diff
@@ -410,6 +519,7 @@
 
 	bus-width = <4>;
 	cap-sd-highspeed;
+	sd-uhs-sdr104;
```
那这个是不是也是多提交进来的？这个应该会导致某些sd卡无法重启吧？

CodeChenL replied:
这个节点是网卡，提速用的，没留意到，所以没分割提交

CodeChenL committed:
arm64: dts: amlogic: enable device-tree overlay support

Add the '-@' DTC option for the Amlogic devices. This option
populates the 'symbols' node that contains all the necessary symbols
for supporting device-tree overlays on these devices.

Reference commit: 3cdba27

Signed-off-by: Jiali Chen <chenjiali@radxa.com>

CodeChenL committed:
arm64: dts: amlogic: add meson-g12b-radxa-zero-2pro

To maintain compatibility with older versions of radxa-pkg/u-boot-latest.
Use a symbolic link to meson-g12b-radxa-zero2 instead of copying for easier tracking and modification.

Signed-off-by: Jiali Chen <chenjiali@radxa.com>

CodeChenL committed:
arm64: dts: meson: radxa-zero2: enable the npu node

Signed-off-by: Jiali Chen <chenjiali@radxa.com>

CodeChenL committed:
drm: panel: jd9365da: modify timing of radxa,display-8hd-ad002

Compatible with amlogic series SOC MIPI DSI interface.
And tested with QCOM platform.

Signed-off-by: Jiali Chen <chenjiali@radxa.com>

RadxaYuntian commented on arch/arm64/boot/dts/amlogic/Makefile:
```diff
@@ -36,6 +40,7 @@ dtb-$(CONFIG_ARCH_MESON) += meson-g12b-odroid-n2-plus.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-g12b-odroid-n2.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-g12b-odroid-n2l.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-g12b-radxa-zero2.dtb
+dtb-$(CONFIG_ARCH_MESON) += meson-g12b-radxa-zero-2pro.dtb
```
最好加个注释

RadxaStephen commented:
Merged.
